### ISSUE TITLE
[Enhancement] Enum to Boolean Converter

### ISSUE THREAD

**Issue by WebDucer** [xamarin/XamarinCommunityToolkit#371]
## Summary

It is often needed to activate / deactivate controls dependent on an Enum value.

## Intended Use Case

```xml
<conv:EnumToBoolConverter x:Key="EnumConverter" />
...
<Button
  IsEnabled="{Binding WorkflowState, Converter={StaticResource EnumConverter}, ConverterParameter={x:Static enum:WorflowState.Staging}}"
  Text="Deploy" />
```

```xml
<conv:EnumToBoolConverter x:Key="EnumConverter" Invert="True" />
...
<Button
  IsVisible="{Binding WorkflowState, Converter={StaticResource EnumConverter}, ConverterParameter={x:Static enum:WorflowState.Finished}}"
  Text="Next" />
```

### As configuration with multiple states

```xml
<conv:EnumsToBoolConverter x:Key="EnumConverter" Invert="True">
  <conv:EnumsToBoolConverter.Values>
    <enum:WorkflowState.Commit />
    <enum:WorkflowState.Build />
    <enum:WorkflowState.Staging />
    <enum:WorkflowState.QA />
  </conv:EnumsToBoolConverter.Values>
</conv:EnumsToBoolConverter>
...
<Button
  IsVisible="{Binding WorkflowState, Converter={StaticResource EnumConverter}}"
  Text="Finish" />
```

## Who Will Do The Work?

- [x] I am willing to take this on myself

**Comment by AndreiMisiukevich** [comment]
**EnumToBoolConverter** ?

it's definitely a good idea.

But how about considering to not stick to "Enum" at all?

StringToBoolConverter or "ValueToBoolConverter" (where value can be whatever: enum, string, int etc.)

**Comment by WebDucer** [comment]
@AndreiMisiukevich I like the idea. Will see, how to do this type safe (I like type safty 😏 ).

**Comment by AndreiMisiukevich** [comment]
Also, I guess we should avoid the "Invert" property. We can build two converters instead

- MatchValueConverter
- NotMatchValueConverter

But I am not sure about the name for such kind converter

CC @sthewissen @Char0394 @jfversluis

**Comment by WebDucer** [comment]
@AndreiMisiukevich How about this naming?
- ValueToBoolConverter
- InvertedValueToBoolConverter
- ValuesCollectionToBoolConverter
- InvertedValuesCollectionToBoolConverter

**Comment by pictos** [comment]
I think this a good candidate for `Multi-Binding`.

**Comment by WebDucer** [comment]
@pictos If the binding used only one time, yes, multibinding could match the needs. If you need the same condition(s) multiple times, multi binding is too verbose and not type safe.

Multi binding is more for use cases, where you are dependent on multiple (changable) values.

**Comment by jfversluis** [comment]
OK so where are we with this? @AndreiMisiukevich @pictos ? You have any more input? 

I like the idea, could be useful!

**Comment by AndreiMisiukevich** [comment]
@WebDucer are you interested in contributing this converter?

**Comment by WebDucer** [comment]
@AndreiMisiukevich Yes. Wait only for a "GO" 😉

**Comment by WebDucer** [comment]
@AndreiMisiukevich We use such a converter internally since 3 years. So I have only to cleanup and make a PR (incl. 100% tests).

**Comment by pictos** [comment]
@WebDucer so go for it! I'll set this issue as `in-progress`, please us know if something happens and you can't work on it anymore.

**Comment by WebDucer** [comment]
@AndreiMisiukevich Hello. I try to execute the unit tests locally, but get always the following compilation error:

Schweregrad | Code | Beschreibung | Projekt | Datei | Zeile | Unterdrückungszustand
-- | -- | -- | -- | -- | -- | --
Fehler | CS0229 | Mehrdeutigkeit zwischen "Resource.Id.snackbar_text" und "Resource.Id.snackbar_text" | Xamarin.CommunityToolkit (MonoAndroid10.0) | c:\src\CommunityToolkit\Xamarin.CommunityToolkit\Views\Snackbar\SnackBar.android.cs | 27 | Aktiv

Is there something, that I have to configure, before I can execute the tests? Or are the tests not executable in VS (Windows)?

**Comment by WebDucer** [comment]
@AndreiMisiukevich Switching to English version of VS helped :/ Don't know why.

**Comment by AndreiMisiukevich** [comment]
Implemented in #925
